h/static/scripts/group-service.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * @ngdoc service
+ * @name  group
+ *
+ * @description
+ * Get and set the UI's currently focused group.
+ */
+'use strict';
+
+// @ngInject
+function group(session) {
+  // The currently focused group. This is the group that's shown as selected in
+  // the groups dropdown, the annotations displayed are filtered to only ones
+  // that belong to this group, and any new annotations that the user creates
+  // will be created in this group.
+  var focusedGroup;
+
+  // The public "group" is always available, regardless of what real groups (if
+  // any) the user is a member of.
+  var publicGroup = {
+    name: 'Public',
+    // We need a sentinel value to identify the public group (which is not
+    // really a group). '__public__' won't clash with any of the hashids that
+    // we use to identify real groups because the hashids never contain _.
+    hashid: '__public__'
+  };
+
+  // Return the list of available groups.
+  var groups = function() {
+    return [publicGroup].concat(session.state.groups || []);
```

Comment:
Hmm. Are you saying the server-side (`session.py`) we should send the public group to the client? Or are you saying that the client-side (`session.js`) would insert this?

Currently the public group doesn't exist server-side / the server doesn't know anything about it. It's just a fake group used in the client-side code to deal with the fact that we have this _Public_ item in the groups dropdown but that isn't really a group.

I'm starting to doubt my decision now. I decided that annotations with no group / belonging to the public group should just have no group field, as this made the server-side code simpler (see discussion in commit message). But now I'm realising that if there were a public group server-side that was always returned in the session and that every annotation belonged to (unless it was in some other group), that might make the _client-side_ code simpler.